sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.messaging.eventhubs.checkpointstore.jedis;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JsonSerializer;
+import com.azure.core.util.serializer.JsonSerializerProviders;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import com.azure.messaging.eventhubs.CheckpointStore;
+import com.azure.messaging.eventhubs.models.Checkpoint;
+import com.azure.messaging.eventhubs.models.PartitionOwnership;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.Jedis;
+
+/**
+ * Implementation of {@link CheckpointStore} that uses Azure Redis Cache, specifically Jedis.
+ */
+public class JedisRedisCheckpointStore implements CheckpointStore {
+    private static final ClientLogger LOGGER = new ClientLogger(JedisRedisCheckpointStore.class);
+    private final JedisPool jedisPool;
+//    private final JacksonAdapter jacksonAdapter = new JacksonAdapter();
+    private static final JsonSerializer DEFAULT_SERIALIZER = JsonSerializerProviders.createInstance(true);
+    JedisRedisCheckpointStore(JedisPool jedisPool) {
+        this.jedisPool = jedisPool;
+
+    }
+
+    /**
+     * This method returns the list of partitions that were owned successfully.
+     *
+     * @param requestedPartitionOwnerships List of partition ownerships from the current instance
+     * @return Null
+     */
+    @Override
+    public Flux<PartitionOwnership> claimOwnership(List<PartitionOwnership> requestedPartitionOwnerships) {
+        return null;
+    }
+
+    /**
+     * This method returns the list of checkpoints from the underlying data store, and if no checkpoints are available, then it returns empty results.
+     *
+     * @param fullyQualifiedNamespace The fully qualified namespace of the current instance of Event Hub
+     * @param eventHubName The Event Hub name from which checkpoint information is acquired
+     * @param consumerGroup The consumer group name associated with the checkpoint
+     * @return Null
+     */
+    @Override
+    public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        ObjectSerializer serializer = DEFAULT_SERIALIZER;
+
+        try (Jedis jedis = jedisPool.getResource()) {
+            Set<String> members = jedis.smembers(prefix);
+            if (members.isEmpty()) {
+                return Flux.error(new IllegalArgumentException()); // no checkpoints to list
```

Comment:
The behaviour here contradicts the javadoc expectation for this method a few lines above. See: https://github.com/Azure/azure-sdk-for-java/pull/29590/files#diff-1044389012002b4bb7b97998ea2d781b0dfd5355433e44c5397f9a1b7e9c256cR51

> and if no checkpoints are available, then it returns empty results.

Reply:
Thanks! That makes sense, I am now returning a Flux generated from an empty Stream. Would that address this?